radix-engine/src/engine/interpreters/native_interpreter.rs

```diff
@@ -102,12 +102,13 @@ impl<N: NativeInvocation> Resolver<N> for NativeResolver {
                 // TODO: Move this logic into kernel
                 let resolved_receiver =
                     if let Some((derefed, derefed_lock)) = deref.deref(receiver)? {
+                        call_frame_update.node_refs_to_copy.insert(receiver);
+                        call_frame_update.node_refs_to_copy.insert(derefed);
```

Comment:
Why include this? I think this will possibly break abstractions and some invariants.

Reply:
Vaguely remember this was due to visibility loss, but will double check again.

Reply:
Yes, and if this is actually a requirement can you add a TODO comment to figure out how to remove that deref in the future